src/DynamoCore/Core/CommandLineRunner.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo;
+using ProtoCore;
+using Dynamo.Models;
+using System.Xml;
+using System.Threading;
+using System.IO;
+
+namespace Dynamo.Core
+{
+    public struct CommandLineArguments
+    {
+        internal static CommandLineArguments FromArguments(string[] args)
```

Comment:
AFAIK `DynamoCore` probably should not have any knowledge of `CommandLineArguments`.  Move this code (translated to the Command Line Parser equivalent) to `DynamoSandbox` in a file called `Options.cs`.